### PR TITLE
feat: Add firmware level Thermocycler emulator

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add firmware level Thermocycler emulator

# Changelog

- Add to Dockerfile to create prod and dev thermocycler images
- Add build and run commands to entrypoint.sh

# Review requests

None

# Risk assessment

None, just adding an image that nothing depends on

### CODE REVIEW

**Comment by DerekMaggio** [comment]
Here is a compose file that you can override `emulation_system/resources/docker/docker-compose.yaml` with.
The inside VM run `./opentrons-emulation em prod --detached` or `./opentrons-emulation em dev --detached`
```yaml
version: '3.8'
services:
  thermocycler-prod:
    image: thermocycler-prod:latest
    build:
      context: .
      target: thermocycler-prod
    container_name: thermocycler-prod
    network_mode: host
    tty: true
    command: ["--stdin"]
```

File: emulation_system/resources/docker/Dockerfile [hunk 1]
```diff
@@ -156,6 +160,9 @@ ENV OPENTRONS_HARDWARE "ot3-firmware-echo"
 FROM cpp-base as heater-shaker-dev
 ENV OPENTRONS_HARDWARE "heater-shaker"
 
+FROM cpp-base as thermocycler-dev
+ENV OPENTRONS_HARDWARE "heater-shaker"
```
Comment (fsinapi) [inline emulation_system/resources/docker/Dockerfile:164]: Should this be `"thermocycler"`?

**Review by fsinapi — CHANGES_REQUESTED** [review]
Looks good, just the one comment 👍

File: emulation_system/resources/docker/Dockerfile (hunk 1 above)
Comment (DerekMaggio) [reply emulation_system/resources/docker/Dockerfile]: Yes it should. Thanks. https://github.com/Opentrons/opentrons-emulation/pull/28/commits/ddc91e0045ff1b7846abfc769b7df4a1d46ec921

**Review by fsinapi — APPROVED** [review]

**Pull request merged**